File: libs/parse/tests/fixtures/java/Hello.java
// Minimal Java fixture used by libs/parse tests. Covers package,
// imports, sealed types, records, pattern matching for instanceof,
// switch expressions, generics, and lambdas — the Java 17+ paths
// most production code we'll scan exercises.

package euxis.sample;

import java.util.List;
import java.util.Optional;

public sealed interface Outcome<T> permits Outcome.Ok, Outcome.Err {
    record Ok<T>(T value) implements Outcome<T> {}
    record Err<T>(String error) implements Outcome<T> {}
}

public final class Hello {

    public record Greeting(String subject, int reps) {
        public String render() {
            return "Hello, " + subject + "!";
        }
    }

    public static int add(int a, int b) {
        return a + b;
    }

    public static Outcome<Greeting> make(String subject) {
        if (subject == null || subject.isBlank()) {
            return new Outcome.Err<>("subject required");
        }
        return new Outcome.Ok<>(new Greeting(subject, 1));
    }

    public static void main(String[] args) {
        var outcome = make("world");
        var summary = switch (outcome) {
            case Outcome.Ok<Greeting> ok -> ok.value().render();
            case Outcome.Err<Greeting> err -> "error: " + err.error();
        };
        var count = add(40, 2);
        List.of(1, 2, 3).forEach(i -> System.out.println(i + ": " + summary + " (" + count + ")"));
        Optional.of(summary).ifPresent(System.out::println);
    }
}

File: libs/parse/tests/fixtures/typescript/hello.ts
// Minimal TypeScript fixture used by libs/parse tests. Exercises
// generics, union types, interface vs type alias, satisfies, const
// assertions, exhaustiveness via never — the paths most production
// TS we'll scan exercises.

interface Greeting<T extends string> {
    readonly subject: T;
    readonly reps: number;
}

type Outcome<T> = { ok: true; value: T } | { ok: false; error: string };

class GreetingFactory {
    static create<T extends string>(subject: T): Outcome<Greeting<T>> {
        if (subject.length === 0) {
            return { ok: false, error: 'subject required' };
        }
        return { ok: true, value: { subject, reps: 1 } satisfies Greeting<T> };
    }
}

const add = (a: number, b: number): number => a + b;

function render<T extends string>(g: Greeting<T>): string {
    return `Hello, ${g.subject}!`;
}

const outcome = GreetingFactory.create('world');
if (outcome.ok) {
    const count = add(40, 2);
    for (let i = 0; i < count; ++i) {
        console.log(`${i}: ${render(outcome.value)}`);
    }
} else {
    console.error(outcome.error);
}
